Plugins: Update plugin signing copy + docs (#34716)
grafana/grafana: docs/sources/plugins/plugin-signatures.md

Plugin signature verification (signing) is a security measure to make sure plugins haven't been tampered with. Upon loading, Grafana checks to see if a plugin is signed or unsigned when inspecting and verifying its digital signature.

> At startup, Grafana verifies the signatures of every plugin in the plugin directory. If a plugin is unsigned, then Grafana won't load or start it. You can see the result of this verification for each plugin by navigating to **Configuration** -> **Plugins**.

Reviewer: ```suggestion
At startup, Grafana verifies the signatures of every plugin in the plugin directory. If a plugin is unsigned, then Grafana does not load nor start it. To see the result of this verification for each plugin, navigate to **Configuration** > **Plugins**.
```

Grafana also writes an error message to the server log:

```bash

> WARN[05-26|12:00:00] Some plugin scanning errors were found     errors="plugin '<plugin id>' is unsigned, plugin '<plugin id>' has an invalid signature"

Reviewer: ```suggestion
WARN[05-26|12:00:00] Some plugin scanning errors were found     errors="plugin '<plugin id>' is unsigned, plugin '<plugin id>' has an invalid signature"
```
What is up with the formatting here?

```

> If you're a plugin developer and want to know how to sign your plugin, refer to [Sign a plugin]({{< relref "../developers/plugins/sign-a-plugin.md" >}}).

Reviewer: ```suggestion
If you are a plugin developer and want to know how to sign your plugin, refer to [Sign a plugin]({{< relref "../developers/plugins/sign-a-plugin.md" >}}).
```

Plugin signature verification (signing) is a security measure to make sure plugins haven't been tampered with. Upon loading, Grafana checks to see if a plugin is signed or unsigned when inspecting and verifying its digital signature.

At startup, Grafana verifies the signatures of every plugin in the plugin directory. If a plugin is unsigned, then Grafana won't load or start it. You can see the result of this verification for each plugin by navigating to **Configuration** -> **Plugins**.

Grafana also writes an error message to the server log:

```bash
WARN[05-26|12:00:00] Some plugin scanning errors were found     errors="plugin '<plugin id>' is unsigned, plugin '<plugin id>' has an invalid signature"
```

> > **Note:** All Grafana Labs authored backend plugins, including Enterprise plugins, are signed.

Reviewer 2: This seems like an odd place for this statement. Isn't it assumed that we sign our own plugins? And if we're going to say it, I would recommend it be up in the prose, not in a note. Perhaps a paragraph leading in to the one in line 21.

Reviewer 3: I agree. It feels like it should be implied that we sign our own plugins. Would it make more sense to invert this statement to list the Grafana Labs authored plugins that aren't (but should be) signed?

Author: I'd be happy to completely omit it if that works too?

>Would it make more sense to invert this statement to list the Grafana Labs authored plugins that aren't (but should be) signed?

Are there many of these? And will they exist like that for a long time?

|Community|<p>Community plugins have dependent technologies that are open source and not for profit.</p><p>Community plugins are published in the official Grafana catalog, and are available to the Grafana community.</p>|
|Commercial|<p>Commercial plugins have dependent technologies that are closed source or commercially backed.</p><p>Commercial Plugins are published on the official Grafana catalog, and are available to the Grafana community.</p>|

## Allow unsigned plugins

We strongly recommend that you don't run unsigned plugins in your Grafana installation. If you're aware of the risks and you still want to load an unsigned plugin, refer to [Configuration]({{< relref "../administration/configuration.md#allow_loading_unsigned_plugins" >}}).

If you've allowed loading of an unsigned plugin, then Grafana writes a warning message to the server log:

```bash

> WARN[06-01|16:45:59] Running an unsigned plugin   pluginID=<plugin id>

Reviewer: ```suggestion
WARN[06-01|16:45:59] Running an unsigned plugin pluginID=<plugin id>
```
